instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/decorator/BaseDecorator.java

```diff
@@ -29,19 +32,26 @@
 import java.lang.reflect.Method;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
-import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ExecutionException;
 
 @Deprecated
 public abstract class BaseDecorator {
 
-  private static final ClassValue<SpanNames> SPAN_NAMES =
-      new ClassValue<SpanNames>() {
-        @Override
-        protected SpanNames computeValue(Class<?> type) {
-          return new SpanNames(getClassName(type));
-        }
-      };
+  private static final QualifiedClassNameCache CLASS_NAMES =
```

Comment:
Can we stick with weak map for this?

Reply:
Why would you use `WeakMap` in preference to `ClassValue`?